src/app/models/save.rb

```diff
@@ -0,0 +1,10 @@
+class Save < ApplicationRecord
+  validates :function, uniqueness: { scope: :user }
+
+  belongs_to :user
+  belongs_to :function
+
+  def self.from_param(function_id)
+    find_by(function_id: function_id)
```

Comment:
this abstraction is incorrect, we use `from_param` as the inverse of `to_param` and as this model doesn't override `#to_param` then this method should be removed in exchange of using `find_by(function: @function)` in the controller